### ISSUE TITLE
plugin/jenkins: improve loading speed of build list

### ISSUE THREAD

**Issue by davegallant** [backstage/backstage#3300]
As noted in the [limitations](https://github.com/backstage/backstage/tree/ad364bdf575a891ee43a0b49ff8cc1046f0f0bee/plugins/jenkins#limitations) in the jenkins plugin, it is very slow loading a project with a large build history.

## Expected Behavior

The first page of builds should load faster and not make so many http requests to Jenkins.

## Current Behavior

It currently makes a new request for each job, and each build in the list, before the first page loads.

## Possible Solution

Rather than retrieving all builds, retrieve the last n (n=100?) builds. The last n builds would likely provide enough context in most cases.

This data might be able to be fetched in a single request such as `https://<JENKINS_HOST>/job/<JOB_NAME>/api/json?tree=jobs[name,url,builds[number,result,duration,url]]`

## Steps to Reproduce

1. Follow the [setup](https://github.com/backstage/backstage/tree/ad364bdf575a891ee43a0b49ff8cc1046f0f0bee/plugins/jenkins#setup) in the plugin README
2. Add a component that maps to a jenkins job with a large build history
3. Wait a long time before the CI table populates

## Context

Any component with a lot of Jenkins builds can take a significant time to load.

## Your Environment

- NodeJS Version (v12): v12.19.0

**Comment by adamdmharvey** [comment]
I looked at this briefly. Since this plugin actually wraps an existing Node.js Package, it's currently limited by that package. (this plugin doesn't call the raw API)

You're correct, in that the passing of the JSON tree XPATH filter should do it, something along the lines of `=tree[name]{0,10}` which would collect the first 0 through 10 entries.  Would then need to tie in the pageSize/numPage to that call, though I didn't have a lot of luck with this. (mostly my learning how the paging in the table component works to be honest!)

**Comment by Fox32** [comment]
I did some minor changes here: #3553 On top of the many requests, the plugin was executing them twice.

About the filtering: the used Jenkins NPM package support passing the `tree` parameter. I played a bit around with it. While it is possible to load all jobs & build in a single request (instead of doing a separate request each time) this solution has the disadvantage of loading a huge amount of data and is equal slow. Limiting the builds to the last 100 seems to be already done by default according to the Jenkins documentation. I would suggest to limit it a lot further, like around 5 requests. I don't think that the use case of seeing historical builds is important.
